Hestia/app/src/main/java/hestia/backend/DeviceListRetrieverTask.java

```diff
@@ -35,37 +35,41 @@ public DeviceListRetrieverTask() {
      */
     @Override
     protected ArrayList<Device> doInBackground(Void... voids) {
-
         String devicesPath = this.cic.getPath() + "devices/";
         URL url = null;
         HttpURLConnection urlConnection = null;
         ArrayList<Device> devices = null;
         try {
             url = new URL(devicesPath);
             urlConnection = (HttpURLConnection) url.openConnection();
+            urlConnection.setReadTimeout(2000);
+            urlConnection.setConnectTimeout(2000);
```

Comment:
Yes. normally it is done in less than a second under all circumstances. However, I cannot force my internet connection to be slightly worse than it is right now in order to properly check this limit. I believe however that it is sufficient.